esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -385,8 +393,12 @@ const char *CActivityInfo::getStatusServerTypeName(WsSMCStatusServerType type)
 
 bool CActivityInfo::findQueueInStatusServer(IPropertyTree* serverStatusRoot, const char* serverName, const char* queueName)
 {
+#ifndef _CONTAINERIZED
     VStringBuffer path("Server[@name=\"%s\"]", serverName);
     Owned<IPropertyTreeIterator> it(serverStatusRoot->getElements(path.str()));
+#else
+    Owned<IPropertyTreeIterator> it(serverStatusRoot->getElements("Server"));
```

Comment:
I'm not clear on why in base-metal it's filtering, and in container mode it isn't ?

Reply:
The function here is to check whether a queue is found in the Status Servers. In a base-metal, the filter (serverName: ThorMaster, HThorServer, etc) is used to narrow down the check. In a non-base-metal, the filter does not work anymore since we do not use the ThorMaster.

Reply:
>In a non-base-metal, the filter does not work anymore since we do not use the ThorMaster.

@wangkx - should be the same as bare-metal, afaik.

Reply:
I think we discussed that before. In non-base-metal, the /Status/Servers does not have the Server which @name is ThorMaster. So, it will not work if the filter is "Server[@name=\"ThorMaster\"]"  ?

Reply:
@wangkx - The code is the same, all Thor master (in future ThorManager) add a Server[@name="ThorMaster"].
Please try it, with multiple jobs, but I think it should work in same way as before in this respect.

Where it differs and we may have talked before, is that unlike bare-metal - where there'll only ever be exactly 1 @thorname e.g. on a multi-thor cluster in bare-metal, they define e.g. 6 thor's all with unique names, e.g. "thor200_20_a",, "thor200_20_b", ... "thor200_20_f".

There will only ever be one of each @thorname, but all will have Server[@name="ThorMaster"] to categorize their type, so you the pre-existing filter by that type in the service code, should still work.

Reply:
Ok, I will try it. I did have the problem before...

Reply:
@jakesmith this is from my dali (my code is based on 7.12.x from Jan. 12):

<Server cluster="thor"
         mpport="7197"
         name="HThorServer"
         node="10.1.28.209"
         queue="thor.thor"
         started="2021-01-14T18:14:25"/>

I did not see any /Status/Servers/Server with @name="ThorMaster".

Reply:
@wangkx - were there thor jobs actually running at the time?

Reply:
No running at the time. I will switch back to this project and check it at a running time.

Reply:
There will be no thor entry (of any kind) when no jobs are running, because in the K8s version, Thor is dynamically launched, multiple Thor's up to ,maxJobs/maxGraphs, when the job is finished, the Thor instance shuts down.

Reply:
If no thor entries when no jobs are running, I should define out the whole findQueueInStatusServer() for non-base-metal. The findQueueInStatusServer() checks the cluster queue entries (including thor queue entries). If an entry is not found, ESP returns 'cluster QUEUENAME not listening for workunits'. For non-base-metal, the error is not valid anymore.